### DISCUSSION TITLE
npm should use stderr for errors

### DISCUSSION THREAD

**Discussion by exx8** [npm/feedback#584] [CLI] [answered]
Whenever npm prints error, it uses stdout. This is wrong. stdout should be used for standard output, for a regular run of the program.
The error should always be printed via stderr, and there many benefits of that, to name a few:

1. It easier for a programmed execution to detect errors (just see if something was printed in stderr, no string stuff) .
2. Many shells support a customized unique way to print error, we should use that instead of using the current universal style (print red ERR!) which might be incompatible for some systems.
3. for even a minute operations, sometimes the error just takes over because all the traceback, rendering the successful results effective invisible (especially in shells that are limited in the number of printed lines).

**Comment by MylesBorins** [comment, marked as answer]
Hey @exx8 thanks for the suggestion!

To make proposals for larger changes (especially breaking ones like this) the appropriate next step is to kick off an issue over at the RFC repo to start the planning / review process.

https://github.com/npm/rfcs

Thanks!
